dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/utils/jaeger/JaegerOtelContainer.java

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.internal.utils.jaeger;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * A container for the Jaeger trace server
+ * <p>
+ * Usage:
+ *
+ * <pre>
+ * {@code @ClassName}
+ * public static JaegerOtelContainer jaegerContainer = new JaegerOtelContainer().withLogConsumer(new SimpleLogConsumer(MyTest.class, "jaeger"));
+ * </pre>
+ *  
+ * Configuration is located in publish/files/otel-collector-config.yaml
+ */
+public class JaegerOtelContainer extends GenericContainer<JaegerOtelContainer> {
```

Comment:
Why is this class needed and how is it different from `JaegerContainer`? It's not clear from the Javadoc.

Reply:
Hmm I initially used it as exposing 4318 and 4317 in the Open Telemetry Collector container and the Jaeger container caused issues. I can change this now I'm using a container network instead of referencing the container hosts externally.